Adds Antiques that replace heirlooms and give Heretics more powers, while also making them less meta-able

eeSPee opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request

### Antiques

Adds antiques in maints, as heirlooms and in a specific cargo pack.

### Rituals

Heretics can perform rituals with some antiques, by using them with a Codex in the other hand. They can use them to gain charges and activate their powers.

### Hexes

Antiques have magical powers that confer both positive and negative effects. Anything from healing, to spawning a blank body to attack the target. Anyone can use these items, but only heretics can see what powers they have. Curators can also identify active antiques.

Heirloom antiques have no magical powers.

## Why It's Good For The Game

Makes heretics less meta-able, and not 100% reliant on tears in reality. 
Gives Curators a purpose that somewhat makes sense.
The last tear in reality change reduced the number of tears, and was an unintended nerf to heretics.

## Changelog
:cl:
add: antiques, 
add: antiques now replace default heirlooms like zippo lighter, cards and 
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

eeSPee committed:
Create antiques.dm

eeSPee committed:
WIP

eeSPee committed:
anitiques

eeSPee committed:
redo

ToasterBan commented:
actually if you allow curators to use antiques that will allow curators to have more cool gimmicks

RDS88-beeman commented:
If this gets merged I'm becoming a curator main.

eeSPee commented:
> actually if you allow curators to use antiques that will allow curators to have more cool gimmicks

But then they would have to be balanced around that.

utilbird commented on code/game/objects/items/antiques.dm:
```diff
@@ -0,0 +1,201 @@
+			to_chat(target,"<span class='warning'>You suddenly cannot talk!</span>")
+		if (GOD_PARALIZE)
+			for(var/obj/item/bodypart/organ in target.bodyparts)
+				if(organ.body_part == LEG_RIGHT || organ.body_part == LEG_LEFT || organ.body_part == ARM_RIGHT || organ.body_part == ARM_LEFT)
```
```suggestion
				if(organ.body_part & (LEG_RIGHT | LEG_LEFT | ARM_RIGHT | ARM_LEFT))
```

eeSPee committed:
Merge branch 'master' into heretix_sacs

eeSPee committed:
icons

eeSPee committed:
Merge branch 'heretix_sacs' of https://github.com/eeSPee/BeeStation-Hornet into heretix_sacs

eeSPee commented:
Alright, fixed conflicts, the code and added some icons. Placeholder or not, you decide:
![image](https://user-images.githubusercontent.com/8404976/120628768-ecbd7d80-c419-11eb-8799-4d06cb93e15a.png)

eeSPee committed:
cooldown + flavortext

eeSPee committed:
Update antiques.dm

eeSPee committed:
Update antiques.dm

eeSPee committed:
revise examinecode

bloons3 commented on code/game/objects/items/antiques.dm:
```diff
@@ -0,0 +1,209 @@
+				if (GOD_FORCE)
+					boon += "pushing the target away."
+				if (GOD_PARALIZE)
+					boon += "paralizing the target."
```
spelling

eeSPee replied:
Fixed

bloons3 commented on code/game/objects/items/antiques.dm:
```diff
@@ -0,0 +1,209 @@
+			to_chat(target,"<span class='warning'>Your crucifix protects you against [user]'s hex!</span>")
+		else 
+			infuse_blessing(user,target)
+		last_use = world.time + 300 SECONDS
```
these have no cost other than only being able to use them every 5 minutes? (Also use 5 MINUTES define instead of seconds)

eeSPee replied:
Yes; figured it's easier to make them balanced if they have some sort of cooldown.

eeSPee committed:
Update antiques.dm

utilbird commented on code/game/objects/items/antiques.dm:
```diff
@@ -0,0 +1,209 @@
+		if (3)
+			name = "Porcelain Doll"
+			desc = "A precious porcelain doll resembling an harlequin."
+			icon_state = "adoll"
```
use lowercase names

utilbird commented on code/game/objects/items/antiques.dm:
```diff
@@ -0,0 +1,209 @@
+/obj/item/artifact/attack_self(mob/living/user)
+	. = ..()
+	if (deity != NO_GOD || inUse)
+		return .
```
`.` is the default return value
```suggestion
		return
```

utilbird commented on code/game/objects/items/antiques.dm:
```diff
@@ -0,0 +1,209 @@
+	else 
+		infuse_blessing(user,user)
+	inUse = FALSE
+	return .
```
```suggestion
```

utilbird commented on code/game/objects/items/antiques.dm:
```diff
@@ -0,0 +1,209 @@
+	desc = "An antique item of some sort..."
+	icon = 'icons/obj/fluff.dmi'
+	icon_state = "paperstack"
+	var/inUse = FALSE
```
nitpick but in_use would be more consistent

eeSPee replied:
Something like this?

eeSPee committed:
Update code/game/objects/items/antiques.dm

Co-authored-by: TheFakeElon <59686430+TheFakeElon@users.noreply.github.com>

eeSPee committed:
Update code/game/objects/items/antiques.dm

Co-authored-by: TheFakeElon <59686430+TheFakeElon@users.noreply.github.com>

eeSPee committed:
Update antiques.dm

utilbird commented on code/game/objects/items/antiques.dm:
```diff
@@ -0,0 +1,208 @@
+				var/datum/antagonist/heretic/ritualist = user.mind.has_antag_datum(/datum/antagonist/heretic)
+				var/result = "[godname] grants you their blessing!"
+				if (!activated)
+					result += ". The [src] is now blessed"
```
This'll turn `blessing!` into `blessing!.`
